### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by Quadry01**

### CODE REVIEW

File: app/LoginPage/page.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import { FcGoogle } from "react-icons/fc";
+import { createUserWithEmailAndPassword, signInWithPopup } from "firebase/auth";
+import { auth, provider } from "../../FirebaseConfig/Firebase";
+
+function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  // REDIRECT USER AFTER SIGNIN
+  const RedirectUser = (e) => {
+    window.location.replace("ProductPage");
```
Comment (ajaps) [inline app/LoginPage/page.js:13]: Nice.

TIL

File: app/LoginPage/page.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+"use client";
+import React, { useState } from "react";
+import { FiShoppingCart } from "react-icons/fi";
+import { FcGoogle } from "react-icons/fc";
+import { createUserWithEmailAndPassword, signInWithPopup } from "firebase/auth";
+import { auth, provider } from "../../FirebaseConfig/Firebase";
+
+function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
```
Comment (ajaps) [inline app/LoginPage/page.js:10]: [Question]
Why do you have to save the email/password?

You should never save users' passwords locally.

File: app/LoginPage/page.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  return (
+    <div className="flex justify-center my-20  ">
+      <div className="bg-cyan-900 w-1/3 h-96 p-24 shadow-lg text-center">
+        <FiShoppingCart className="m-auto text-white text-8xl" />
+        <span className=" text-white text-2xl"> Super Shopping</span>
+      </div>
+
+      <div className="w-1/3 h-96 bg-yellow-100 shadow-lg">
+        <form className="pl-20 pt-8">
```
Comment (ajaps) [inline app/LoginPage/page.js:46]: Oh, I see why you saved the email and password.

You don't have to do that though, you can easily access the info when the form is submitted.

See this as a guide:  https://contactmentor.com/login-form-react-js-code/

File: app/ProductPage/page.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    getDocs(productCollection)
+      .then((res) => {
+        const products = res.docs.map((doc) => ({
+          data: doc.data(),
+          id: doc.id,
+        }));
+        setProductList(products);
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (ajaps) [inline app/ProductPage/page.js:27]: [Suggestion]
You should send the error message to users via notification/pop-up

Guide: https://dev.to/cesareferrari/how-to-display-error-messages-in-a-react-application-3g48

**Review by ajaps — COMMENTED** [review]
I took a first pass.  When you make the updates put them in new `commits`.

I'm hoping Seun will be able to take a look at the FE code(React best practises and styling)

File: app/LoginPage/page.js (hunk 2 above)
Comment (Quadry01) [reply app/LoginPage/page.js:10]: I did not store the login credentials locally, i used the state to pass the email and password to firebase

File: app/ProductPage/page.js (hunk 4 above)
Comment (Quadry01) [reply app/ProductPage/page.js:27]: I understand sir.

**Pull request merged**